python/lsst/daf/persistence/posixStorage.py

```diff
@@ -308,13 +352,38 @@ def read(self, butlerLocation):
 
         return results
 
+    def butlerLocationExists(self, location):
+        """Implementaion of PosixStorage.exists for ButlerLocation objects."""
+        storageName = location.getStorageName()
+        if storageName not in ('BoostStorage', 'FitsStorage', 'PafStorage',
+                               'PickleStorage', 'ConfigStorage', 'FitsCatalogStorage'):
```

Comment:
it's not going to grow, at least not much. We are planning to add a pluggable system for storageName functionality.